src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -313,15 +316,28 @@ private async Task GenerateRequest(EntryComponent entry, int order, Cancellation
             AddHeaderIfNeeded(HeaderNames.IfNoneMatch, entry.Request.IfNoneMatch, httpContext);
             AddHeaderIfNeeded(KnownHeaders.IfNoneExist, entry.Request.IfNoneExist, httpContext);
 
-            if (requestMethod == HTTPVerb.POST ||
-                requestMethod == HTTPVerb.PUT)
+            if (requestMethod == HTTPVerb.POST
+                || requestMethod == HTTPVerb.PUT)
             {
                 httpContext.Request.Headers.Add(HeaderNames.ContentType, new StringValues(KnownContentTypes.JsonContentType));
 
                 var memoryStream = new MemoryStream(_fhirJsonSerializer.SerializeToBytes(entry.Resource));
                 memoryStream.Seek(0, SeekOrigin.Begin);
                 httpContext.Request.Body = memoryStream;
             }
+#if !STU3
+            // Allow JSON Patch to be an encoded Binary in a Bundle (See: https://chat.fhir.org/#narrow/stream/179166-implementers/topic/Transaction.20with.20PATCH.20request)
+            else if (
+                requestMethod == HTTPVerb.PATCH &&
+                string.Equals(KnownResourceTypes.Binary, entry.Resource?.TypeName, StringComparison.Ordinal) &&
+                entry.Resource is Binary binaryResource && string.Equals(KnownMediaTypeHeaderValues.ApplicationJsonPatch.ToString(), binaryResource.ContentType, StringComparison.OrdinalIgnoreCase))
```

Comment:
>entry.Resource is Binary binaryResource  [](http://example.com/codeflow?start=16&length=40)

[https://thumbs.gfycat.com/CloseGoodAmericanriverotter-size_restricted.gif](https://thumbs.gfycat.com/CloseGoodAmericanriverotter-size_restricted.gif)

Do we need to do same thing with binary for other methods as well, or it's special hack for PATCH?

Is this wild spec interpretation, or FHIR spells this somewhere?

Reply:
Yes on both accounts.
See https://chat.fhir.org/#narrow/stream/179166-implementers/topic/Transaction.20with.20PATCH.20request and https://smilecdr.com/docs/fhir_repository/updating_data.html

Its not great or pretty, but in some cases where a conversion routine is going from hl7v2 => FHIR, PATCH operations are required inside bundles (I believe most importantly in transaction bundles), this "hack/workaround" unblocks this. It seems like the only near-term solution we can offer until we implement FHIR Path Patch which uses the Parameters object.